File: app/src/main/java/com/dicoding/moviecataloguerv/database/SearchDao.java
package com.dicoding.moviecataloguerv.database;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import com.dicoding.moviecataloguerv.model.Search;

import java.util.List;

@Dao
public interface SearchDao {

    @Insert
    void insert(Search search);

    @Delete
    void delete(Search search);

    @Query("SELECT * FROM query_search ORDER BY id DESC")
    LiveData<List<Search>> getAllSearch();

    @Query("SELECT `query` FROM query_search WHERE `query`=:query")
    String getSearch(String query);

    @Query("DELETE FROM query_search")
    void deleteAllData();
}

File: app/src/main/java/com/ryanrvldo/moviecatalogue/data/local/database/SearchDao.kt
package com.ryanrvldo.moviecatalogue.data.local.database

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Query
import com.ryanrvldo.moviecatalogue.data.model.Search

@Dao
interface SearchDao : BaseDao<Search> {

    @Query("SELECT * FROM query_search ORDER BY id DESC")
    fun getSearchQueries(): LiveData<List<Search>>

    @Query("SELECT `query` FROM query_search WHERE `query`=:query")
    suspend fun selectSearchQuery(query: String): String

    @Query("DELETE FROM query_search")
    suspend fun deleteSearchQueries()

}